cloudmarker/events/azrdbmssslenforcementevent.py

```diff
@@ -0,0 +1,104 @@
+"""Microsoft Azure RDBMS SSL Enforcement Event.
+
+This module defines the :class:`AzRDBMSSLEnforcementEvent` class that
+identifies RDBMS servers which have SSL connection enforcement
+disabled. This plugin works on the  properties found in the ``ext``
+bucket of ``rdbms`` records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class AzRDBMSSSLEnforcementEvent:
+    """Az RDBMS SSL enforcement event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`AzRDBMSSLEnforcementEvent`."""
+
+    def eval(self, record):
+        """Evaluate Azure RDBMS servers for SSL connection enforcement.
+
+        Arguments:
+            record (dict): A RDBMS record.
+
+        Yields:
+            dict: An event record representing an RDBMS server where SSL
+            connection enforcement is disabled
+
+        """
+        com = record.get('com', {})
+        if com is None:
+            return
+
+        if com.get('record_type') != 'rdbms':
+            return
+
+        if com.get('cloud_type') != 'azure':
+            return
+
+        ext = record.get('ext', {})
+        if ext is None:
+            return
+
+        if ext.get('record_type') not in ['mysql_server',
+                                          'postgresql_server']:
+            return
+
+        if ext.get('ssl_connection_enabled', False) is False:
+            yield from _get_azure_rdbms_ssl_enforcement_event(
+                com, ext)
+
+    def done(self):
+        """Perform cleanup work.
+
+        Currently, this method does nothing. This may change in future.
+        """
+
+
+def _get_azure_rdbms_ssl_enforcement_event(com, ext):
+    """Generate event for SSL enforcement disabled.
+
+    Arguments:
+        com (dict): RDBMS record `com` bucket
+        ext (dict): RDBMS record `ext` bucket
+    Returns:
+        dict: An event record representing RDBMS server with
+        SSL connection enforcement disabled
+
+    """
+    friendly_cloud_type = util.friendly_string(com.get('cloud_type'))
+    friendly_rdbms_type = util.friendly_string(ext.get('record_type'))
+
+    reference = com.get('reference')
+    description = (
+        '{} {} {} has SSL connection enforcement disabled'
+        .format(friendly_cloud_type, friendly_rdbms_type, reference)
+    )
+    recommendation = (
+        'Check {} {} {} and enable SSL connection enforcement'
+        .format(friendly_cloud_type, friendly_rdbms_type, reference)
```

Comment:
Please end `recommendation` and `description` messages with a full-stop. The intention here is to write human-friendly complete sentences that could be picked up from the event and used directly in emails if necessary.

I see that the same update is required in the disk encryption event plugins. A separate pull request can be created for those plugins.

Reply:
Corrected  `recommendation` and `description` so that they end with a full-stop.